app/jobs/appointment_reminders/send_reminder_job.rb

```diff
@@ -0,0 +1,98 @@
+class AppointmentReminders::SendReminderJob
+  include Rails.application.routes.url_helpers
+  include Sidekiq::Worker
+
+  sidekiq_options queue: :high
+
+  def perform(reminder_id)
+    reminder = AppointmentReminder.includes(:appointment, :patient).find(reminder_id)
+    if reminder.status != "scheduled"
+      report_error("scheduled appointment reminder has invalid status")
+    else
+      send_message(reminder)
+    end
+  end
+
+  private
+
+  def send_message(reminder)
+    notification_service = NotificationService.new
+
+    begin
+      response = if communication_type == "missed_visit_whatsapp_reminder"
+        notification_service.send_whatsapp(
+          phone_number(reminder.patient),
+          appointment_message(reminder),
+          callback_url
+        )
+      else
+        notification_service.send_sms(
+          phone_number(reminder.patient),
+          appointment_message(reminder),
+          callback_url
+        )
+      end
+
+      Communication.create_with_twilio_details!(
+        appointment: reminder.appointment,
+        appointment_reminder: reminder,
+        twilio_sid: response.sid,
+        twilio_msg_status: response.status,
+        communication_type: communication_type
+      )
+    rescue Twilio::REST::TwilioError => e
```

Comment:
I would really prefer to move this error handling to the NotificationService, but I'm leaving it alone for now.